test/context_test.rb

```diff
@@ -73,6 +82,21 @@ def test_sampling_priority
     end
   end
 
+  def test_origin
```

Comment:
In order to avoid accruing any more debt, we should probably add this new test to RSpec rather than to Minitest. Modifying an existing Minitest is fine, but just want to avoid a conversion of something that can be already written as RSpec from the get-go.